package.json

```diff
@@ -110,7 +110,7 @@
     "lodash.shuffle": "^4.2.0",
     "loglevel": "^1.4.1",
     "luxon": "^1.8.2",
-    "metamask-inpage-provider": "^1.3.0",
+    "metamask-inpage-provider": "^2.0.1",
```

Comment:
This is a major semver (breaking) change! Do we know for sure this is the provider we want to bring in?

Reply:
Upon review, the only thing that's changed with `metamask-inpage-provider` since `1.3.0` is package updates: https://github.com/MetaMask/metamask-inpage-provider/commits/master

I we want to, I can probably still revert without issue.

Reply:
The changelog only notes v5 as adding babelify@10: https://github.com/MetaMask/json-rpc-engine/blob/master/CHANGELOG.md

Looks like the breaking change of 4.x.x was making next() calls trigger even when there’s an error. So next handlers should check for errors, ideally. I’m unclear if this would break anything. https://github.com/MetaMask/json-rpc-engine/commit/6de511afbd03ccef4550ea43ff4010b7d7a84039

it’s a big open question of “do our middlewares behave well when their next() functions are called after errors?" We could take the chance, but we should QA heavily and make sure we have the long community QA period, and also probably roll out cautiously.

Reply:
Aaron says he verified our middlewares were behaving well, so this looks safe to me.